src/main/java/frc/robot/commands/drivetrain/TurnToAngle.java

```diff
@@ -0,0 +1,49 @@
+package frc.robot.commands.drivetrain;
+
+import static frc.robot.Constants.DriveConstants.turnD;
+import static frc.robot.Constants.DriveConstants.turnI;
+import static frc.robot.Constants.DriveConstants.turnP;
+import static frc.robot.Constants.DriveConstants.turnRateToleranceDegPerS;
+import static frc.robot.Constants.DriveConstants.turnToleranceDeg;
+
+import edu.wpi.first.wpilibj.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.PIDCommand;
+import frc.robot.subsystems.DriveTrain;
+
+/**
+ * A command that will turn the robot to the specified angle. Copied from wpilib
+ */
+public class TurnToAngle extends PIDCommand {
+
+  /**
+   * Turns to robot to the specified angle.
+   *
+   * @param targetAngleDegrees The angle to turn to
+   * @param driveTrain         The drive subsystem to use
+   */
+  public TurnToAngle(double targetAngleDegrees, DriveTrain driveTrain) {
+    super(
+        new PIDController(turnP, turnI, turnD),
+        // Close loop on heading
+        driveTrain::getHeading,
+        // Set reference to target
+        targetAngleDegrees,
+        // Pipe output to turn robot
+        output -> driveTrain.worldOfTanksDrive(0, 0, output),
```

Comment:
Unsure if this is necessary, but you use DriveTrain.getInstance in every other auto when you need the DriveTrain object. You might want to use that method here to keep consistency. You may want to create a class local DriveTrain object to store this instance in instead.
```suggestion
        output -> DriveTrain.getInstance.worldOfTanksDrive(0, 0, output),
```